### DISCUSSION TITLE
Extract deeply nested (more than 1 level) plan

### DISCUSSION THREAD

**Discussion by mnunezdm** [forcedotcom/cli#2359] [Ideas]
**Is your feature request related to a problem? Please describe.**

Extracting data with the `sfdx force:data:tree:export` only allows to extract one level of nested elements. I think it would be great to extract deeply nested plans like for example: `Account->Case->Task`

I managed to manually extract the data in two steps, manually edit the data plan, and insert them using the `sfdx force:data:tree:import -p`

**Describe the solution you'd like**

Ability to, for example, retrieve a list of queries and make the plan automatically for example:

```
$ sfdx force:data:tree:export -p -q "SELECT Id, Name (SELECT Origin FROM Cases) FROM Account"
                                    "SELECT Id, (SELECT Id FROM Tasks), (SELECT Id FROM Events) FROM Case"
```

**Describe alternatives you've considered**

NA

**Additional context**

This is the workaround I did: https://salesforce.stackexchange.com/questions/240042/sfdx-forcedatatreeexport-beyond-1-level-deep/255711#255711

**Comment by KertulPL** [comment]
Yep! Same problem :) we need 5 level deep

**Comment by sfscgjohnson** [comment]
Agree this is needed.

**Comment by gaelCangy** [comment]
Having the same issue here. Would be awesome to have up to 5 level :). Also taking into consideration that the query could reference the same object more than 2 time like contact -> contact -> contact or something similar would be great

**Comment by WillieRuemmele** [comment]
https://github.com/forcedotcom/cli/issues/127

**Comment by mshanemc** [comment]
The API supports this, it's a bug in the CLI's implementation.  Converting it back to an issue.
